radixdlt/src/main/java/com/radixdlt/middleware2/network/TestEventCoordinatorNetwork.java

```diff
@@ -115,37 +111,40 @@ public void setReceivingDisable(ECPublicKey validatorId, boolean disable) {
 		}
 	}
 
-	private int getRandomLatency() {
-		if (minimumLatency == maximumLatency) {
-			return minimumLatency;
-		} else {
-			return minimumLatency + rng.nextInt(maximumLatency - minimumLatency + 1);
-		}
-	}
-
 	public EventCoordinatorNetworkSender getNetworkSender(ECPublicKey forNode) {
-		Consumer<MessageInTransit> sendMessageSink = message -> {
-			if (!sendingDisabled.contains(forNode)) {
-				if (preserveOrder) {
-					orderedMessageBuffer.push(message);
+		final Map<ECPublicKey, Long> lastTimestamps = new ConcurrentHashMap<>();
+
+		final Consumer<MessageInTransit> sendMessageSink = message -> {
+			if (sendingDisabled.contains(forNode)) {
+				return;
+			}
+
+			final int nextLatency;
+			if (message.target.equals(forNode)) {
+				nextLatency = 0;
+			} else {
+
+				final long curTime = System.currentTimeMillis();
+				final Long lastTimestamp = lastTimestamps.get(message.target);
```

Comment:
Ha, funny enough I considered doing it this way as well before deciding on the queue-based approach. My reasoning was that it would be harder to ensure preserved ordering in a threadsafe way without the queue, with the added limitation that there is no easy way to view the currently pending messages (which I've found useful for debugging). 

Specifically, in my limited understanding, I'm not confident the current implementation is thread-safe because the code between reading the timestamp (line 128) and writing the next timestamp back (line 136) is not synchronized.